docs/architecture-decision-records/0020-aggregation-for-tdr-preingest.md

```diff
@@ -0,0 +1,40 @@
+# 20. Aggregation for TDR Preingest
+
+**Date:** 2024-08-14
+
+## Context
+
+The first source to ingest into DR2 was parsed court documents transferred via TDR and parsed by TRE. The upstream workflow for this type of record enforces a 1-1 relationship between a transfer consignment and a record. We're now ready to create a preingest route for standard TDR transfers.
+
+Standard TDR transfers contain multiple files within a consignment; the count and size of the files is at the discretion of the transferring user. While the Preservation System ingest process is most efficient when ingesting batches, as opposed to individual files, the size of the batches are unlikely to match the user decisions upstream of DR2; we want to design a preingest process for TDR to be most efficient for the Preservation System.
+
+## Decision
+
+We will implement an aggregation process as part of our TDR Preingest workflow, we will request TDR notify us of new records for ingest with a message per record to allow us to aggregate. Our aggregation process will use the same AWS services as our ingest process - SQS, Lambda, Step Functions, and DynamoDB.
+
+### Aggregation Design
+
+![TDR Aggregation Diagram](/docs/images/adr/0020/tdr-aggregation-diagram.png)
+
+We will extend our usage of the existing dr2-ingest-lock table (created for [`0016-generic-ingest-idempotency.md`](./0016-generic-ingest-idempotency.md)) to act as a lock table within our aggregation. We will add a `groupId` attribute and GSI on this. And a `message` attribute to store the JSON string of the message received from TDR.

```

Comment:
1. There doesn't need to be a full stop/new sentence after "on this"
2. Could you add what a `groupId` is and why we need it?

Reply:
2. Added
> The `groupId` will be a unique identifier shared by all messages aggregated into a single group.